tests/Opus/Search/Common/IndexerTest.php

```diff
@@ -0,0 +1,961 @@
+
+		$class  = new ReflectionClass( 'Opus_SolrSearch_Index_Indexer' );
+		$method = $class->getMethod( 'sendSolrXmlToServer' );
+		$method->setAccessible( true );
+
+		$method->invoke( $this->indexer, $xml );
+	}
+
+	/**
+	 * Regression test for OPUSVIER-2417
+	 */
+	public function testFulltextVisibilityIsConsideredInFacetForFrontdoorVisibleFulltext() {
+		$this->markTestSkipped( 'not supported in Opus_Search API' );
+
+/*		$doc = new Opus_Document( $this->document_id );
+		$doc->setServerState( 'published' );
+		$doc->setLanguage( 'eng' );
+
+		$file = $doc->addFile();
+		$file->setTempFile( APPLICATION_PATH . '/tests/fulltexts' . DIRECTORY_SEPARATOR . 'test.pdf' );
+		$file->setPathName( 'test.pdf' );
+		$file->setVisibleInFrontdoor( '1' );
+
+		$doc->store();
+
+		$class  = new ReflectionClass( 'Opus_SolrSearch_Index_Indexer' );
+		$method = $class->getMethod( 'getSolrXmlDocument' );
+		$method->setAccessible( true );
+
+		$xml       = $method->invoke( $this->indexer, $doc );
+		$xmlString = $xml->saveXML();
+		$this->assertContains( '<field name="has_fulltext">true</field>', $xmlString );
+		$this->assertNotContains( '<field name="has_fulltext">false</field>', $xmlString );
+		$this->assertContains( '<field name="fulltext_id_success">' . $file->getId() . ':' . $file->getRealHash( 'md5' ) . '</field>', $xmlString );
+		$this->assertNotContains( '<field name="fulltext_id_failure">', $xmlString );
+
+		$path = $this->files_dir . DIRECTORY_SEPARATOR . $doc->getId();
+		unlink( $path . DIRECTORY_SEPARATOR . 'test.pdf' );
+		rmdir( $path );*/
+	}
+
+	/**
+	 * Regression test for OPUSVIER-2417
+	 */
+	public function testFulltextVisibilityIsConsideredInFacetForFrontdoorInvisibleFulltext() {
+		$this->markTestSkipped( 'not supported in Opus_Search API' );
+
+/*		$doc = new Opus_Document( $this->document_id );
+		$doc->setServerState( 'published' );
+		$doc->setLanguage( 'eng' );
+		$file = $doc->addFile();
+		$file->setTempFile( APPLICATION_PATH . '/tests/fulltexts' . DIRECTORY_SEPARATOR . 'test.pdf' );
+		$file->setPathName( 'test.pdf' );
+		$file->setVisibleInFrontdoor( '0' );
+		$doc->store();
+
+		$class  = new ReflectionClass( 'Opus_SolrSearch_Index_Indexer' );
+		$method = $class->getMethod( 'getSolrXmlDocument' );
+		$method->setAccessible( true );
+
+		$xml       = $method->invoke( $this->indexer, $doc );
+		$xmlString = $xml->saveXML();
+		$this->assertContains( '<field name="has_fulltext">false</field>', $xmlString );
+		$this->assertNotContains( '<field name="has_fulltext">true</field>', $xmlString );
+		$this->assertNotContains( '<field name="fulltext_id_success">', $xmlString );
+		$this->assertNotContains( '<field name="fulltext_id_failure">', $xmlString );
+
+		$path = $this->files_dir . DIRECTORY_SEPARATOR . $doc->getId();
+		unlink( $path . DIRECTORY_SEPARATOR . 'test.pdf' );
+		rmdir( $path );*/
+	}
+
+	/**
+	 * Regression test for OPUSVIER-2417
+	 */
+	public function testFulltextVisibilityIsNotConsideredInFacet() {
```

Comment:
This one and the two above it are regression tests. How can I be sure that the new code does not reintroduce the same problems? As I have admitted before I did't write the old search code and am likely to overlook something.